Components/AddDestinations/AddDestinations.js

```diff
@@ -1,10 +1,36 @@
 import React, { useState, useEffect } from 'react';
-import { StyleSheet, Text, TextInput, TouchableOpacity, View, ScrollView } from 'react-native'
+import { StyleSheet, Text, TextInput, TouchableOpacity, View, ScrollView } from 'react-native';
+import { createNewDestination } from '../../apiCalls';
+import DatePicker from '../DatePicker/DatePicker';
+
 const AddDestinations = ({ navigation, route }) => {
-  // const { tripId } = route.params
+  const { tripId, handleTripsFetch } = route.params;
   let [destinationLocation, setDestinationLocation] = useState('');
   let [destinationStartDate, setDestinationStartDate] = useState('');
-  let [destinationEndDate, setDestinationEndDate] = useState('')
+  let [destinationEndDate, setDestinationEndDate] = useState('');
+  let [newDestination, setNewDestination] = useState([]);
+
+  //Use conditional rendering to notify the user that they are adding a destination #${tripdestinationSet.length} - requires some props reformation
+
+  const handleDestinationSubmit = () => {
+    createNewDestination(tripId, destinationLocation, destinationStartDate, destinationEndDate)
+      .then(returnedTripData => {
+        setNewDestination(returnedTripData);
+        navigation.navigate('Trips')
+        handleTripsFetch()
+      })
+      .catch(error => {
+        console.log(error)
+      });
+      resetInputs();
+  }
+
```

Comment:
Thanks for debugging the fetching/navigation flow. I know it got a little hairy there, but it's nice and smooth now!